### ISSUE TITLE
[BUG] Git url regex matches any windows path in projects.run()

### ISSUE THREAD

**Issue by kokoff** [mlflow/mlflow#2824]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [x] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.7.2
- **Python version**: Python 3.7.3
- **npm version, if running the dev UI**: N/A
- **Exact command to reproduce**: python -c "from mlflow import projects;projects.run('C:\asd')"

### Describe the problem
Describe the problem clearly here. Include descriptions of the expected behavior and the actual behavior.

Git uri regex matches absolute Windows path starting with drive (e.g. C:\..). As a result running projects.run('C:\...') results in the error:
```
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v origin
  stderr: 'fatal: 'C:/asd' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.'
```

The git uri regex seemse to be too broad, matching any path containing `:`
```
_GIT_URI_REGEX = re.compile(r"^[^/]*:")
```

### Code to reproduce issue
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

```
from mlflow import projects
projects.run('C:\\asd')
```

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

```
>>> from mlflow import projects
>>> projects.run('C:\\asd')
2020/05/14 09:45:26 INFO mlflow.projects: === Fetching project from C:\asd into C:\Users\USER_NAME\AppData\Local\Temp\tmpciv1cczr ===
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\miniconda3\lib\site-packages\mlflow\projects\__init__.py", line 288, in run
    use_conda=use_conda, storage_dir=storage_dir, synchronous=synchronous, run_id=run_id)
  File "D:\miniconda3\lib\site-packages\mlflow\projects\__init__.py", line 104, in _run
    work_dir = _fetch_project(uri=uri, force_tempdir=False, version=version)
  File "D:\miniconda3\lib\site-packages\mlflow\projects\__init__.py", line 342, in _fetch_project
    _fetch_git_repo(parsed_uri, version, dst_dir)
  File "D:\miniconda3\lib\site-packages\mlflow\projects\__init__.py", line 382, in _fetch_git_repo
    origin.fetch()
  File "D:\miniconda3\lib\site-packages\git\remote.py", line 790, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress)
  File "D:\miniconda3\lib\site-packages\git\remote.py", line 674, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "D:\miniconda3\lib\site-packages\git\cmd.py", line 412, in wait
    raise GitCommandError(self.args, status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
  cmdline: git fetch -v origin
  stderr: 'fatal: 'C:/asd' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.'
```

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [x] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

**Comment by smurching** [comment]
@kokoff makes sense, would be happy to review a PR that scopes the git URL regex more appropriately! Adding the `important-longterm` label but let me know if you're interested in contributing and I'll update to `important-soon`

**Comment by theyorubayesian** [comment]
Is this issue resolved in 1.22.0? I still get this error when running mlflow against windows path
